sw/device/lib/testing/test_framework/ottf_common.ld

```diff
@@ -239,5 +228,69 @@ SECTIONS {
     *(.freertos.heap)
   } > ram_main
 
+  /**
+   * Non-volatile scratch and counter areas for tests.
+   *
+   * These sections are meant to be used by tests to store data that should
+   * persist across resets. Since `opentitantool` assumes that the signed
+   * region of an image extends until the end, these sections are placed after
+   * all other sections and are NOLOAD so that they don't appear in the binary
+   * and therefore don't end up being covered by the signature.
+   *
+   * Note that these sections are placed in `eflash` instead of `ottf_flash` so
+   * that they don't interfere with tests that use address translation.
+   */
+   _non_volatile_scratch_size = DEFINED(no_ottf_nv_scratch) ? 0 : 20480;
+   _non_volatile_scratch_end = ORIGIN(ottf_flash) + _ottf_size;
+   _non_volatile_scratch_start = _non_volatile_scratch_end - _non_volatile_scratch_size;
+
+   _non_volatile_counter_size = DEFINED(no_ottf_nv_counter) ? 0 : 2048;
+   _non_volatile_counter_flash_words = _non_volatile_counter_size / 8;
+
+   _non_volatile_counter_0_end = _non_volatile_scratch_start;
+   _non_volatile_counter_0_start = _non_volatile_counter_0_end - _non_volatile_counter_size;
+
+   _non_volatile_counter_1_end = _non_volatile_counter_0_start;
+   _non_volatile_counter_1_start = _non_volatile_counter_1_end - _non_volatile_counter_size;
+
+   _non_volatile_counter_2_end = _non_volatile_counter_1_start;
+   _non_volatile_counter_2_start = _non_volatile_counter_2_end - _non_volatile_counter_size;
+
+  /**
+   * Non-volatile scratch area in flash.
+   */
+  .non_volatile_scratch _non_volatile_scratch_start (NOLOAD) : ALIGN(2048) {
+    KEEP(*(.non_volatile_scratch))
+    KEEP(*(.non_volatile_scratch.*))
+    . = _non_volatile_scratch_size;
+  } > ottf_flash
+
+
+  /**
+   * Non-volatile counter area in flash.
```

Comment:
Sure, but instead of re-purposing the counter sections, defining a type + API to be used in the generic scratch area would probably be better for clarity.